code/orphaned_procs/dbcore.dm

```diff
@@ -122,9 +122,15 @@ DBQuery
 
 DBQuery/proc/Connect(DBConnection/connection_handler) src.db_connection = connection_handler
 
+DBQuery/proc/warn_execute()
+	if(!Execute())
+		usr << "<span class='danger'>A SQL error occured during this operation, check the server logs.</span>"
+
 DBQuery/proc/Execute(sql_query=src.sql,cursor_handler=default_cursor)
 	Close()
-	return _dm_db_execute(_db_query,sql_query,db_connection._db_con,cursor_handler,null)
+	. = _dm_db_execute(_db_query,sql_query,db_connection._db_con,cursor_handler,null)
+	if(!.)
+		diary << "\[[time_stamp()]]SQLERROR: [ErrorMsg()] | Query used: [sql]"
```

Comment:
Please make a new log_sql proc in the matter of log_game or log_admin, and call the category SQL not SQLERROR so that it can be used for all sql related logging.

(You could even make a log_sqlerror() that calls log_sql() so that errors have the format of SQL: ERROR:)

I want to keep logging inside those procs because at some point we may want to do a logging database and having everything there would be helpful for that.

On that note, Maybe add a arg for disabling the logging of errors, i foresee some point when we may have to run sql queries entirely to see if they fail, and if we do do database logs, the insert query would need to not log otherwise you'd get a infinite loop if it errored out